dali/kernels/slice/slice_cpu.h

```diff
@@ -28,31 +29,238 @@ namespace kernels {
 
 namespace detail {
 
-template <typename OutputType, typename InputType, int Dims>
+/**
+ * @brief Fills output with nchannel values repeatedly
+ */
+template <typename T>
+void Fill(T *output, const T *fill_values, int64_t npixels, int64_t nchannels) {
+  int64_t n = npixels * nchannels;
+  int64_t i = 0;
+  for (; i < nchannels; i++)
+    output[i] = fill_values[i];
+  for (; i < n; i++)
+    output[i] = output[i - nchannels];
+}
+
+static inline std::tuple<int64_t, int64_t, int64_t> CalcPadCopyExtents(int64_t anchor,
+                                                                       int64_t in_extent,
+                                                                       int64_t out_extent) {
+  int64_t pad_before = std::max(0l, std::min(out_extent, -anchor));
```

Comment:
This code is not portable - `long` does not need to be 64-bit.
```suggestion
  int64_t pad_before = std::max<int64_t>(0, std::min(out_extent, -anchor));
```